src/data/nat/enat.lean

```diff
@@ -269,12 +269,17 @@ lemma pos_iff_one_le {x : enat} : 0 < x ↔ 1 ≤ x :=
 enat.cases_on x (by simp only [iff_true, le_top, coe_lt_top, ← @nat.cast_zero enat]) $
   λ n, by { rw [← nat.cast_zero, ← nat.cast_one, enat.coe_lt_coe, enat.coe_le_coe], refl }
 
-noncomputable instance : linear_order enat :=
-{ le_total := λ x y, enat.cases_on x
+instance : is_total enat (≤) :=
+{ total := λ x y, enat.cases_on x
     (or.inr le_top) (enat.cases_on y (λ _, or.inl le_top)
       (λ x y, (le_total x y).elim (or.inr ∘ coe_le_coe.2)
-        (or.inl ∘ coe_le_coe.2))),
+        (or.inl ∘ coe_le_coe.2))) }
+
+noncomputable instance : linear_order enat :=
```

Comment:
Note that we have `lattice.to_linear_order`.

Reply:
I don't think a full lattice is available yet?